lib/bolt_server/base_config.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+require 'hocon'
+require 'bolt/error'
+
+module BoltServer
+  class BaseConfig
+    def config_keys
+      ['host', 'port', 'ssl-cert', 'ssl-key', 'ssl-ca-cert',
+       'ssl-cipher-suites', 'loglevel', 'logfile', 'whitelist',
+       'file-server-conn-timeout', 'file-server-uri']
+    end
+
+    def env_keys
+      ['ssl-cert', 'ssl-key', 'ssl-ca-cert', 'loglevel',
+       'file-server-conn-timeout', 'file-server-uri']
+    end
+
+    def defaults
+      { 'host' => '127.0.0.1',
+        'ssl-cipher-suites' => ['ECDHE-ECDSA-AES256-GCM-SHA384',
+                                'ECDHE-RSA-AES256-GCM-SHA384',
+                                'ECDHE-ECDSA-CHACHA20-POLY1305',
+                                'ECDHE-RSA-CHACHA20-POLY1305',
+                                'ECDHE-ECDSA-AES128-GCM-SHA256',
+                                'ECDHE-RSA-AES128-GCM-SHA256',
+                                'ECDHE-ECDSA-AES256-SHA384',
+                                'ECDHE-RSA-AES256-SHA384',
+                                'ECDHE-ECDSA-AES128-SHA256',
+                                'ECDHE-RSA-AES128-SHA256'],
+        'loglevel' => 'notice',
+        'file-server-conn-timeout' => 120 }
+    end
+
+    def initialize(config = nil)
+      @data = defaults
+      @data = @data.merge(config.select { |key, _| config_keys.include?(key) }) if config
+      @config_path = nil
+      config_keys.each do |key|
+        self.class.send(:define_method, key.tr('-', '_').to_sym) { @data[key] }
```

Comment:
Are these methods even that useful? Can we just remove them?

Reply:
Since we already have `@data` and have a getter for it I decided to just use that, but if there's a better way to access those variables I'm open to it.